src/Chapter1.hs

```diff
@@ -537,7 +543,15 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: Ord a => a -> a -> a -> a
+mid x y z
+  | x <= y && z <= x = x
+  | x <= z && y <= x = x
+  | y <= x && z <= y = y
+  | y <= z && x <= y = y
+  | z <= x && y <= z = z
+  | z <= y && x <= z = z
+  | otherwise = y
```

Comment:
It looks like you can move all these 3 cases under a single otherwise and return `z` in here.
The implementation will be a bit shorter 🙂

Reply:
Good point! I was actually surprised I had to add the otherwise and the cases weren't already exhaustive.

Reply:
As we mention in Chapter Two, the compiler in Haskell is very attentive to the exhaustive pattern-matching. And here it would warn you that `Pattern matching is not exhaustive`, as the guards have quite complicated logic, and the compiler won't be able to prove that it covers all the cases.

So, guards usually always end with `otherwise` :slightly_smiling_face:

Reply:
Ah that makes sense. Guess I was giving the compiler too much credit on checking guards :)